File: ungs-docsys-bff/src/modules/resume-user/dtos/resume-file-request.dto.ts
import {
  IsOptional,
  IsString,
  MaxLength
} from 'class-validator';

export class ResumeFileRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(5000, { message: 'File content must not exceed 5000 characters' })
  fileBinary?: string;
}

File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/ResumeFileRequestDto.java
package com.ungs.docsys.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ResumeFileRequestDto {
    @NotBlank(message = "File content must not be blank")
    @Size(max = 5000, message = "File content must not exceed 5000 characters")
    private String fileBinary;
}
